x/launch/keeper/validator_set.go

```diff
@@ -0,0 +1,64 @@
+package keeper
+
+import (
+	"bytes"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/spn/x/launch/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// CheckValidatorSet check the information about a validator
+// set (used to create an IBC client) are valid
+func (k Keeper) CheckValidatorSet(
+	ctx sdk.Context,
+	launchID uint64,
+	chainID string,
+	validatorSet tmtypes.ValidatorSet,
+) (bool, error) {
+	// check chain ID
+	chain, found := k.GetChain(ctx, launchID)
+	if !found {
+		return false, sdkerrors.Wrapf(types.ErrChainNotFound, "%d", launchID)
+	}
+	if !chain.LaunchTriggered {
+		return false, sdkerrors.Wrapf(types.ErrNotTriggeredLaunch, "%d", launchID)
+	}
+	if chain.GenesisChainID != chainID {
+		return false, sdkerrors.Wrap(types.ErrInvalidGenesisChainID, chainID)
+	}
+
+	// all validators must be present in the launch module and
+	// the total amount of self-delegation from the provided validators
+	// must reach at least 2/3 of the total self delegation for the chain
+	valSetSelfDelegation := 0.0
+	for _, validator := range validatorSet.Validators {
+		valAddr := sdk.AccAddress(validator.Address.Bytes())
+		launchValidator, found := k.GetGenesisValidator(ctx, launchID, valAddr.String())
```

Comment:
I realized we omitted in the specs we have the consensus address and not the validator operator's address. Therefore `valAddr` can't retrieve the validator
Since we only index the validator by the operator address, I think we should with all validators on spn retrieved with `GetAllGenesisValidatorByLaunchID` create an internal map that indexes the consensus pub key of the validator with the full validator or the self delegation to make then this check